### PR TITLE
Increase HTTP endpoint ping logging.

### PR DESCRIPTION
**Pull request by UmanShahzad**
This increases the logging in case of failures, and also in all other cases, to give us more visibility on what's going on with snitch pinging.

To test, install the requirements into a virtualenv, cd to the tests directory and run

    python -m unittest test_sanity_check

### CODE REVIEW

File: files/sanity_check.py [hunk 1]
```diff
@@ -179,16 +180,23 @@ def sanity_check(json_data):
...
-                return True
+                report.append(SanityCheckResult(True, "Attempt {} to contact DMS succeeded.".format(count)))
+                break
+            report.append(SanityCheckResult(True, "Attempt {} to contact DMS returned code "
+                                                  "{}".format(count, response.getcode())))
         time.sleep(delay)
-    return False
+    else:
+        report.append(SanityCheckResult(False, "Couldn't send snitch after {} attempts".format(max_retries)))
+    return report
```
Comment (itsjeyd) [inline files/sanity_check.py:199]: @UmanShahzad Nice approach turning `ping_http_endpoint` into a report-generating function :)

**Review by itsjeyd — APPROVED** [review]
@UmanShahzad Looking good, thanks for this :+1:

- [x] I tested this: Works as advertised in PR description.
- [x] I read through the code
- [x] ~~I checked for accessibility issues~~ N/A
- [x] Includes documentation (is self-documenting :))

File: files/sanity_check.py (hunk 1 above)
Comment (UmanShahzad) [reply files/sanity_check.py:199]: Thanks! :)

**Comment by antoviaque** [comment]
@UmanShahzad Thanks for implementing this : )

**Pull request merged**